src/kingdom/AbstractAnimals.java

```diff
@@ -0,0 +1,44 @@
+package kingdom;
+
+
+
+public abstract class AbstractAnimals
+{
+    private static int maxId=0;
+    protected int id;
+    protected String name;
+    protected int yearDiscovered;
+    
```

Comment:
These variables don't even have to be `protected` - just use the `private` access modifier for them, since the `super` constructor in the subclasses can still use their values anyway.